src/bun.js/api/bun/subprocess.zig

```diff
@@ -1421,15 +1479,18 @@ pub const Subprocess = struct {
             jsc_vm.tick();
             jsc_vm.eventLoop().autoTick();
         }
-
         // subprocess.wait(true);
         const exitCode = subprocess.exit_code orelse 1;
         const stdout = subprocess.stdout.toBufferedValue(globalThis);
         const stderr = subprocess.stderr.toBufferedValue(globalThis);
         subprocess.finalizeSync();
 
         const sync_value = JSC.JSValue.createEmptyObject(globalThis, 4);
-        sync_value.put(globalThis, JSC.ZigString.static("exitCode"), JSValue.jsNumber(@intCast(i32, exitCode)));
+        if (subprocess.abort_reason != .zero) {
+            sync_value.put(globalThis, JSC.ZigString.static("exitCode"), subprocess.abort_reason);
```

Comment:
@colinhacks @Electroid what should it return here?

Reply:
Thoughts?
```ts
spawn({ ... }): Subprocess;
spawn({ ..., signal }): AbortableSubprocess;
```

```ts
type Subprocess = {
  readonly exitCode: number;
}

type AbortableSubprocess = {
  readonly exitCode: number;
  readonly aborted: false;
} | {
  readonly exitCode: null;
  readonly aborted: true;
  readonly reason: unknown;
};
```

Reply:
An alternative:

```ts
type Subprocess = {
  // if aborted, exitCode is 124, which matches:
  // https://manpages.courier-mta.org/htmlman1/timeout.1.html
  readonly exitCode: number;
  // if aborted, non-null
  readonly signal: AbortSignal | null;
};
```